src/convert/datetime.jl

```diff
@@ -6,8 +6,14 @@ rcopy(::Type{Date}, s::Ptr{RealSxp}) = rcopy(Date, s[1])
 rcopy(::Type{DateTime}, s::Ptr{RealSxp}) = rcopy(DateTime, s[1])
 
 rcopy(::Type{Date}, x::Float64) = Date(Dates.UTInstant(Dates.Day((isnan(x) ? 0 : x) + 719163)))
-rcopy(::Type{DateTime}, x::Float64) =
-    DateTime(Dates.UTInstant(Dates.Millisecond(((isnan(x) ? 0 : x) + 62135683200) * 1000)))
+function rcopy(::Type{DateTime}, x::Float64)
+    ms = ((isnan(x) ? 0 : x) + 62135683200) * 1_000
+    if !isinteger(ms)
+        @warn "Precision lost in conversion to DateTime"
```

Comment:
Though R users are accustomed to warnings, they're rather unusual for Julia. Perhaps it would make sense for this to be either an error or a `@debug` message? Any idea what the impact would be of the precision loss in the worst case?

Reply:
Hmmmm.... erroring is in some sense the current behavior, but we could add a more meaningful error message that the user needs to round to millisecond precision first.

R essentially allows for infinite precision here (I couldn't find a documented precision), but Julia restricts this to milliseconds. So you're never more than 1ms off, but whether that matters depends a lot on application domain. Looking at the POSIX roots of this, POSIX mostly cares about seconds, so 🤷. 

I think `@debug` is fine. It's better rounding than what a user would get if they rounded in R, then did the conversion to Julia:
```
R> x <- as.POSIXct('2020-10-09 12:09:46.1234')

R> round(x, "millisecond")
Error in match.arg(units) : 
  'arg' should be one of “secs”, “mins”, “hours”, “days”, “months”, “years”
```

Reply:
Any idea how common it is to have `POSIXct` values in R with meaningful sub-millisecond precision? At any rate, it could be worth calling this out somewhere in the documentation.

Reply:
https://github.com/JuliaInterop/RCall.jl/pull/533/commits/3766f29272bccab282041152d0e6eb94f6fbfbf4